### ISSUE TITLE
Use variable as a key instead of a value

### ISSUE THREAD

**Issue by pedrosimao** [graphql-editor/graphql-zeus#233]
I am trying to do some group GraphQL requests in which the group is determined by a key.
Here is a quick example: (notice the `groupBy` variable)

```
export const cardsGroupSelector = Selectors.query({
  cardsConnection: [
    {
      limit: $`limit`,
      where: $`where`,
    },
    {
      groupBy: {
        [$`groupBy`]: {
          key: true,
          connection: {
            values: {
              id: true,
              name: true,
              checkInStatus: true,
              requestStatus: true,
              coverImage: {
                id: true,
                url: true,
              },
            },
          },
        },
      },
    },
  ],
})
```

graphql-Zeus don't transform this key correctly before sending the request. My final GraphQL request looks like this:

```
{"query":"query($limit:Int, $where:JSON){cardsConnection(limit:$limit,where:$where){groupBy{ZEUS_VAR$groupBy{key connection{values{id name designer season lookNumber priority checkInStatus requestStatus coverImage{id url}}}}}}}","variables":{"where":{"project":{"id":"38"}},"groupBy":"requestStatus","limitCollaborators":3}}
```

As you can notice graphql-zeus send this key `groupBy{ZEUS_VAR$groupBy` instead of this one `requestStatus`.... I am not sure there is a solution for this, since having a key as a variable will probably influence the type returned by the request. Is there any chance that we can find a solution for this?

**Comment by aexol** [comment]
You need to use aliases. This is an incorrect usage. Variables are only for input values in GraphQL

**Comment by pedrosimao** [comment]
Hello @aexol , I tried playing with `aliases` after I got your comment.
I manage to create aliases easily, with `graphql-zeus` but unhappily it does not apply to the described problem/use case.
On the problem I described I don't want to rename a key from the response object, because I don't even know what key I want in the first place. That's why aliases won't apply, and it feels more like a variable. It's just not a graphql variable, but more like a normal javascript variable.

The `groupBy` request can actually group database information depending on a specific key name. So if I want to group db items by their author's name, I should use the key `author` on my request. I am not sure there is a possible solution to this problem.
If there is not I will have to copy and paste the same Selectors `query` multiple times, but with only one different key for each individual query.
